components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/consent/SSOConsentServiceImpl.java

```diff
@@ -664,9 +674,49 @@ private String buildSubjectWithUserStoreDomain(AuthenticatedUser authenticatedUs
 
     protected boolean isConsentForClaimValid(PIICategoryValidity piiCategoryValidity) {
 
+        String consentValidity = piiCategoryValidity.getValidity();
+
+        if (isEmpty(consentValidity)) {
+            return true;
+        }
+
+        List<String> consentValidityEntries = Arrays.asList(consentValidity.split(ConsentSSOConstants
+                                                                                          .CONSENT_VALIDITY_SEPARATOR));
+        for (String consentValidityEntry : consentValidityEntries) {
+            if (consentValidityEntry.toUpperCase().startsWith(CONSENT_VALIDITY_TYPE_VALID_UNTIL)) {
+                String[] validityEntry = consentValidityEntry.split(CONSENT_VALIDITY_TYPE_SEPARATOR, 2);
+                if (validityEntry.length == 2) {
+                    long currentTimeMillis = System.currentTimeMillis();
+                    try {
+                        String validTime = validityEntry[1];
+                        if (isDebugEnabled()) {
+                            String message = String.format("Validity time for PII category: %s is %s.",
+                                                           piiCategoryValidity.getName(), validTime);
+                            logDebug(message);
+                        }
+                        if (isExpiryIndefinite(validTime)) {
+                            return true;
+                        }
+                        long consentExpiryInMillis = Long.parseLong(validTime);
+                        return consentExpiryInMillis > currentTimeMillis;
+                    } catch (NumberFormatException e) {
+                        if (isDebugEnabled()) {
+                            String message = String.format("Cannot parse timestamp: %s. for PII category %s.",
+                                                           consentValidity, piiCategoryValidity.getName());
+                            logDebug(message);
+                        }
+                    }
+                }
+            }
+        }
         return true;
```

Comment:
Is this correct to return true even if something goes wrong ? ex- even if number format exception occurs it will return true

Reply:
Added this on purpose to avoid failing expirations which cannot be interpreted. I guess we can fail `VALID_UNTIL` type of expirations if they cannot be interpreted. WDYT ?

Reply:
In such cases we better to ask consent again.

Reply:
For any unknown expiry type or only for unparsable Authentication `VALID_UNTIL` expiry type ?